WooExpress Trial: Users should not be able to launch sites on WooExpress Trial plan

valterlorran opened the pull request:
#### Proposed Changes

* Adds a nudge on the "Launch site" card to let the users know that to launch the site, they need to upgrade to an eCommerce plan.

<img width="732" alt="Screenshot 2023-01-24 at 17 01 14" src="https://user-images.githubusercontent.com/3376401/214329285-10fa3238-721b-4a92-b032-14b25c22d66a.png">

#### Testing Instructions

* On your site with the WooExpress trial plan, navigate to `/settings/general/{SITE}`
* Scroll down to the "Launch site" card.
* You should see the nudge inviting the user to upgrade to an eCommerce plan.
* You can also click on the nudge, it should take you to the plans page.

Closes #71859

valterlorran committed:
The launch button should be disabled when the is site is on the eCommerce trial plan

valterlorran committed:
Adding the nudge when for ecommerce trail

markbiek commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -583,12 +585,18 @@ export class SiteSettingsFormGeneral extends Component {
 		const btnText = translate( 'Launch site' );
 		let querySiteDomainsComponent;
 		let btnComponent;
+		let eCommerceTrailUpsell;
```
```suggestion
		let eCommerceTrailUpsell;
```

markbiek commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +617,26 @@ export class SiteSettingsFormGeneral extends Component {
 		const LaunchCard = showPreviewLink ? CompactCard : Card;
 
+		if ( isSiteOnECommerceTrial ) {
+			eCommerceTrailUpsell = (
```
```suggestion
			eCommerceTrialUpsell = (
```

markbiek commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +617,26 @@ export class SiteSettingsFormGeneral extends Component {
 			<>
 				<SettingsSectionHeader title={ translate( 'Launch site' ) } />
 				<LaunchCard>
+					{ eCommerceTrailUpsell }
```
```suggestion
					{ eCommerceTrialUpsell }
```

markbiek reviewed (commented):
Functionality works as expected. Just needs a couple of typos fixed.

valterlorran committed:
Fix typo

sixhours commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +617,26 @@ export class SiteSettingsFormGeneral extends Component {
+						icon="info"
+						showDismiss={ false }
+						text={ translate(
+							'In order for everyone to see your store, you just need to subscribe to a eCommerce plan. Subscribe'
```
I think "Subscribe" would make sense as a `<NoticeAction />` linked to the plans page, to make it more clear this notice is click-able, rather than wrapping the whole component in an anchor.

I know this is following the given design, but the wording here feels off to me. It would be good to get another set of eyes on the copy before we add the "String Freeze" label to get it ready for translation.

daledupreez replied:
Agreed on both counts: we should use the `NoticeAction` component for the CTA, and the wording here is a bit awkward.

@verofasulo, could you take a first pass at the wording for us? My thinking is that something like the following would be better:
> You can launch your store by subscribing to a paid eCommerce plan [Subscribe]

or 

> To launch your store, you need to subscribe to a paid eCommerce plan [Subscribe]

daledupreez commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +617,26 @@ export class SiteSettingsFormGeneral extends Component {
 
+		if ( isSiteOnECommerceTrial ) {
+			eCommerceTrialUpsell = (
+				<a href={ `/plans/${ siteSlug }?plan=business-bundle-monthly` }>
```
Shouldn't this be the eCommerce bundle?

valterlorran replied:
I fixed it in my last commit. Thanks for noticing that :)

verofasulo commented:
Hey @daledupreez 👋

> could you take a first pass at the wording for us? 

@FlorianBru submitted a UX copy request to the marketing team. The copy review will be finalized by Jan 23 - Does this work for you? 

In the meantime, this is the suggestion that @FlorianBru made:

![image](https://user-images.githubusercontent.com/98944206/213436161-cd973406-cea4-4f3c-a601-4f55eb9dc486.png)

valterlorran committed:
- Changes the redirect to the ecommerce-bundle-monthly plan
 - Add the NoticeAction component instead of wrapping the Notice with a link
 - Use the text suggested

valterlorran commented:
Thanks, @verofasulo. I applied your suggestion in my last commit.

daledupreez commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +617,33 @@ export class SiteSettingsFormGeneral extends Component {
 		const LaunchCard = showPreviewLink ? CompactCard : Card;
 
+		if ( isSiteOnECommerceTrial ) {
+			const eCommerceTrialUpsellText = translate(
+				'Before you can share your store with the world, you need to {{NoticeAction}}pick a plan{{/NoticeAction}}.',
+				{
+					components: {
+						NoticeAction: (
+							<NoticeAction href={ `/plans/${ siteSlug }?plan=ecommerce-bundle-monthly` } />
+						),
+					},
+				}
+			);
```
Given that we've switched to inline text for the CTA, we should use a more standard `a` tag, and not a `NoticeAction`. If you take a look at the [Notice dev docs](https://wpcalypso.wordpress.com/devdocs/design/notice), the `NoticeAction` is intended for a separate CTA (which we had before), and not an inline link.
```suggestion
			// TODO: flesh this out and wire it up, but we want to be able to understand how often people click on the CTA
			// We should also trigger an event when the notice is rendered - can we do that via a functional wrapper component that uses `useEffect()` to trigger a render based on the incoming props?
			const recordTracksEventForClick = recordTracksEvent( 'calypso_ecommerce_trial_launch_banner_click' );
			const eCommerceTrialUpsellText = translate(
				'Before you can share your store with the world, you need to {{a}}pick a plan{{/a}}.',
				{
					components: {
						a: (
							<a href={ `/plans/${ siteSlug }?plan=ecommerce-bundle-monthly` } onClick={ recordTracksEventForClick }  />
						),
					},
				}
			);
```

daledupreez commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +617,33 @@ export class SiteSettingsFormGeneral extends Component {
+					showDismiss={ false }
+					text={ eCommerceTrialUpsellText }
+					isCompact={ false }
+				></Notice>
```
Minor stylistic note: You can also use `/>` to close off the element.
```suggestion
				/>
```

daledupreez reviewed (commented):
@valterlorran, can you refactor the `NoticeAction` into a simpler component and update the CSS to match?

And can you consider whether you'd like to add Tracks events in this PR or a follow-up? I think we should also handle that work in this PR so we can safely wait for translations to come in.

valterlorran committed:
- Uses a link instead of the NoticeAction
 - Removes the custom style for the NoticeAction in the Notice
 - Adds the tracks events

sixhours commented on client/my-sites/site-settings/form-general.jsx:
```diff
@@ -609,10 +618,38 @@ export class SiteSettingsFormGeneral extends Component {
+					components: {
+						a: (
+							<a
+								href={ `/plans/${ siteSlug }?plan=ecommerce-bundle-monthly` }
```
We should probably use the plan constant (although I recognize this will probably change!) rather than a hard-coded value.

valterlorran replied:
Fixed it :)

valterlorran committed:
Use the plan constant instead of hardcoding it

sixhours reviewed (approved):
Working well! Once the copy is finalized/cleared let's add the String Freeze label to get translations started. :)

daledupreez commented:
@valterlorran and @sixhours, I think we can get this merged and deployed without waiting for the strings to be translated, as no users should see this upsell yet.

a8ci18n commented:
This Pull Request is now available for translation here: https://translate.wordpress.com/deliverables/7771747

Thank you @valterlorran for including a screenshot in the description! This is really helpful for our translators.

a8ci18n commented:
Translation for this Pull Request [has now been finished](https://translate.wordpress.com/deliverables/overview/7771747).

FlorianBru commented:
Just a quick update, because the copy was changed. So the new design and content look like this:
<img width="720" alt="image" src="https://user-images.githubusercontent.com/111048810/220647018-7a44cc31-16c0-4141-9e85-3baabc7a56be.png">
